docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,220 @@
+- Host a remote configuration (using a CDN)
+
+### Remote Configuration
+
+The remote configuration will specify configuration data for the assisted-service such as `DATA_COLLECTION_URL`. This can be cached in a CDN.
+
+The assisted service will get this remote configuration:
+
+1. When the assisted service initially starts
+2. Every time the assisted service will send data it'll get this configuration first
+
+If the assisted service can't get the latest configuration, it'll fallback to the last configuration received or the hard-coded default it was deployed with.
+
+#### Benefits
+1. Future-proof this feature
+    - The configuration is not reliant on the version of the assisted-service deployed
+    - E.g. if the data collection endpoint changes, older versions will be able to send it to the new endpoint without having to update their client
+2. Allows flexibility for our team
+    - We can adjust the configurations on the fly such as where data is sent to
+    - E.g. if the original data collection endpoint is unavailable or if we choose to use a different service later on, we only need to change the configuration endpoint 
+
+#### Drawbacks
+1. Another deployment artifact to update and maintain
+2. Doesn't fully future-proof against the remote configuration's URL changes
+    - Although we can always have the end user change the URL by redeploying the service with the new URL set as an environment variable 
+
+### User Stories
+
+#### Story 1
+
+As an infrastructure operator, I can provision and manage my fleet of clusters knowing that data points are automatically collected and sent to the Assisted Installer team without having to do any additional configuration. I know my data will be safe and secure and the team will only collect data they need to improve the product. 
+
+#### Story 2
+
+As a developer on the assisted installer team, I can analyze the customer data to determine if a feature is worth implementing/keeping/improving. I know that the customer data is accurate and up-to-date. All of the data is parse-able and can be easily tailored to the graphs/visualizations that help my analysis.
+
+#### Story 3
+
+As a product owner, I can determine if the product is moving in the right direction based on the actual customer data. I can help determine which features and bug fixes to prioritize based on the data.
+
+### Implementation Details/Notes/Constraints
+
+Assisted Service changes:
+1. New environment variables to be added:
+    - `DISABLE_DATA_COLLECTION`: If set to `true`, it will disable this feature and thereby allow a customer to opt-out, otherwise the feature is on by default.
+    - `DATA_COLLECTION_URL`: The URL the data will be sent to. (Data format env var?)
+    - `REMOTE_CONFIG_URL`: The URL for the manifest with configuration data for on-prem data collection variables such as `DATA_COLLECTION_URL`
+2. Add data collection and sending logic
```

Comment:
What process does this?

Is this going to be just some kind of fire-and-forget goroutine that gets spun off after we transition to an end state?
Or is it going to be another persistent "worker"-type thread that is constantly running, waiting for these events?

Reply:
I think it'll be more similar to the fire-and-forget goroutine. 

I'm thinking of creating an event_uploader controller that has an upload function. 

So the flow would go something like this:
1. A cluster fails/finishes installation 
2. There's a separate function call to query the DB for events of that cluster
3. Then the event_uploader controller would be called to upload the events (including sanitization and packaging of data)


What do you think?